src/java.base/share/classes/java/nio/channels/spi/AbstractInterruptibleChannel.java

```diff
@@ -89,10 +89,26 @@ public abstract class AbstractInterruptibleChannel
     private final Object closeLock = new Object();
     private volatile boolean closed;
 
+    // invoked if a Thread is interrupted when blocked in an I/O op
+    private final Interruptible interruptor;
+
     /**
      * Initializes a new instance of this class.
      */
-    protected AbstractInterruptibleChannel() { }
+    protected AbstractInterruptibleChannel() {
+        this.interruptor = new Interruptible() {
+            @Override
+            public void interrupt(Thread target) {
+                AbstractInterruptibleChannel.this.trySetTarget(target);
+            }
+            @Override
+            public void postInterrupt() {
+                try {
+                    AbstractInterruptibleChannel.this.close();
```

Comment:
Before the current change, the `Interruptible` implementation in this class used to call `AbstractInterruptibleChannel.this.implCloseChannel();` whereas now it calls `close()`. Is that an intentional change?

Reply:
Yes, this is intentional.